dmoj/urls.py

```diff
@@ -95,230 +94,230 @@ def exception(request):
-        url(r'^/resubmit/(?P<submission>\d+)$', problem.ProblemSubmit.as_view(), name='problem_submit'),
-
-        url(r'^/rank/', paged_list_view(ranked_submission.RankedSubmissions, 'ranked_submissions')),
-        url(r'^/submissions/', paged_list_view(submission.ProblemSubmissions, 'chronological_submissions')),
-        url(r'^/submissions/(?P<user>[\w-]+)/', paged_list_view(submission.UserProblemSubmissions, 'user_submissions')),
-
-        url(r'^/$', lambda _, problem: HttpResponsePermanentRedirect(reverse('problem_detail', args=[problem]))),
-
-        url(r'^/test_data$', ProblemDataView.as_view(), name='problem_data'),
-        url(r'^/test_data/init$', problem_init_view, name='problem_data_init'),
-        url(r'^/test_data/diff$', ProblemSubmissionDiff.as_view(), name='problem_submission_diff'),
-        url(r'^/data/(?P<path>.+)$', problem_data_file, name='problem_data_file'),
-
-        url(r'^/tickets$', ticket.ProblemTicketListView.as_view(), name='problem_ticket_list'),
-        url(r'^/tickets/new$', ticket.NewProblemTicketView.as_view(), name='new_problem_ticket'),
-
-        url(r'^/manage/submission', include([
-            url('^$', problem_manage.ManageProblemSubmissionView.as_view(), name='problem_manage_submissions'),
-            url('^/rejudge$', problem_manage.RejudgeSubmissionsView.as_view(), name='problem_submissions_rejudge'),
-            url('^/rejudge/preview$', problem_manage.PreviewRejudgeSubmissionsView.as_view(),
+    path('', blog.PostList.as_view(template_name='home.html', title=_('Home')), kwargs={'page': 1}, name='home'),
+    path('500/', exception),
+    path('admin/', admin.site.urls),
+    path('i18n/', include('django.conf.urls.i18n')),
+    path('accounts/', include(register_patterns)),
+    path('', include('social_django.urls')),
+
+    path('problems/', problem.ProblemList.as_view(), name='problem_list'),
+    path('problems/random/', problem.RandomProblem.as_view(), name='problem_random'),
+
+    path('problem/<str:problem>', include([
+        path('', problem.ProblemDetail.as_view(), name='problem_detail'),
+        path('/editorial', problem.ProblemSolution.as_view(), name='problem_editorial'),
+        path('/pdf', problem.ProblemPdfView.as_view(), name='problem_pdf'),
+        path('/pdf/<slug:language>', problem.ProblemPdfView.as_view(), name='problem_pdf'),
+        path('/clone', problem.ProblemClone.as_view(), name='problem_clone'),
+        path('/submit', problem.ProblemSubmit.as_view(), name='problem_submit'),
+        path('/resubmit/<int:submission>', problem.ProblemSubmit.as_view(), name='problem_submit'),
+
+        path('/rank/', paged_list_view(ranked_submission.RankedSubmissions, 'ranked_submissions')),
+        path('/submissions/', paged_list_view(submission.ProblemSubmissions, 'chronological_submissions')),
+        path('/submissions/<slug:user>/', paged_list_view(submission.UserProblemSubmissions, 'user_submissions')),
```

Comment:
Let's use `str` for all users instead of `slug`. Pretty sure `\w` matched non-ASCII characters. We have some pretty wild usernames in prod...